src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp

```diff
@@ -142,6 +142,24 @@ void BattlegroundWS::RespawnFlagAfterDrop(TeamId teamId)
     _bgEvents.CancelEvent(BG_WS_EVENT_BOTH_FLAGS_KEPT10);
     _bgEvents.CancelEvent(BG_WS_EVENT_BOTH_FLAGS_KEPT15);
     RemoveAssaultAuras();
+
+    CheckFlagKeeperInArea(teamId == TEAM_ALLIANCE ? TEAM_HORDE : TEAM_ALLIANCE);
+}
+
+void BattlegroundWS::CheckFlagKeeperInArea(TeamId teamId)
+{
+    if (GetStatus() != STATUS_IN_PROGRESS || GetFlagState(teamId) != BG_WS_FLAG_STATE_ON_PLAYER)
+        return;
+
+    uint32 triggerId = teamId == TEAM_ALLIANCE ? 3647 : 3646;
```

Comment:
you could create an enum for these trigger IDs and replace these numbers with names like
```c++
enum WG_TRIGGER {
   CAPTURE_FLAG = 3647,
   TAKE_FLAG = 3646
}
```

Reply:
Done, thanks for the suggestion!